libraries/ms-common/src/x86/uart.c

```diff
@@ -2,9 +2,8 @@
 
 // basic idea: tx is stored in a buffer, interrupt-driven
 // rx is buffered, once a newline is hit or the buffer is full, call rx_handler
-// requires tty0tty to be installed
-// sudo depmod && sudo modprobe tty0tty && sudo chmod 666 /dev/tnt*
-// must be run each time vagrant is started
+// run the command once on vagrant startup
+// socat -d -d pty,raw,echo=0,link=/home/vagrant/s0 pty,raw,echo=0,link=/home/vagrant/s1 & disown
```

Comment:
Can we not require commands to be run on startup? This also makes it impossible to run more than one program without risking running into another. If I want to connect port 1 to port 2 on a program, how do I do that? How would this support running 3 of the same program?